engine/api/src/org/apache/cloudstack/engine/cloud/entity/api/VirtualMachineEntity.java

```diff
@@ -115,6 +115,12 @@ String reserve(DeploymentPlanner plannerToUse, @BeanParam DeploymentPlan plan, E
     boolean stop(String caller) throws ResourceUnavailableException, CloudException;
 
     /**
+     * Stop the virtual machine, optionally force
+     *
+     */
+    boolean stop(String caller, boolean forced) throws ResourceUnavailableException, CloudException;
```

Comment:
@rhtyd since this PR adds new methods, let's not carry it forward.  The implementation of ``forceStop`` would end up passing ``true`` for the ``forced`` flag down the stack, but at least we are not expanding an anti-pattern.